plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/ScaleDeploymentHandler.java

```diff
@@ -52,81 +72,217 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 		IReplicationController rc = getSelectedElement(event, IReplicationController.class);
 		if(rc != null) {
 			scaleUsing(event, rc, rc.getName());
-			return null;
-		}
-		IServiceWrapper deployment = getSelectedElement(event, IServiceWrapper.class);
-		if(deployment != null) {
-			Collection<IResourceWrapper<?, ?>> rcs = deployment.getResourcesOfKind(ResourceKind.REPLICATION_CONTROLLER);
-			if(!rcs.isEmpty()) {
-				//there should be only 1 per deployment, we'll assume this is true
-				rc = (IReplicationController) rcs.iterator().next().getWrapped();
-				scaleUsing(event, rc, deployment.getWrapped().getName());
+		} else {
+			IServiceWrapper deployment = getSelectedElement(event, IServiceWrapper.class);
+			if (deployment != null) {
+				rc = getReplicationController(deployment);
+				if (rc != null) {
+					scaleUsing(event, rc, deployment.getWrapped().getName());
+				}
 			}
 		}
 		return null;
 	}
 	
+	private IReplicationController getReplicationController(IServiceWrapper deployment) {
+		Collection<IResourceWrapper<?, ?>> rcs = deployment.getResourcesOfKind(ResourceKind.REPLICATION_CONTROLLER);
+		if(rcs.isEmpty()) {
+			return null;
+		}
+		//there should be only 1 per deployment, we'll assume this is true
+		return (IReplicationController) rcs.iterator().next().getWrapped();
+	}
+
 	protected void scaleUsing(ExecutionEvent event, IReplicationController rc, String name) {
-		final int replicas = getDesiredReplicas(rc, event);
-		scaleDeployment(event, name, rc, replicas);
+		final int requestedReplicas = getRequestedReplicas(rc, name, event);
+		final int currentReplicas = rc.getDesiredReplicaCount();
+		if (requestedReplicas != -1
+				&& currentReplicas != requestedReplicas) { 
+				if (requestedReplicas == 0
+						&& !showStopDeploymentWarning(name, HandlerUtil.getActiveShell(event))) {
+					return;
+				}
+				scaleDeployment(event, name, rc, requestedReplicas);
+		}
 	}
 	
+	private boolean showStopDeploymentWarning(String name, Shell shell) {
+		MessageDialog dialog = new MessageDialog(shell, 
+				"Stop all deployments?", 
+				OpenShiftCommonImages.OPENSHIFT_LOGO_WHITE_ICON_IMG, 
+				NLS.bind("Are you sure you want to scale {0} to 0 replicas?\nThis will stop all pods for the deployment.", name),
```

Comment:
It seems correct to me but English is only my primary language :)  Alternative might be "Are you certain you want to stop all pods for this deployment"